src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -805,6 +807,7 @@ class JavaValue {
     jint     i;
     jlong    l;
     jobject  h;
```

Comment:
Do we still need jobject after this change?

Reply:
We still use jobject for the arguments. We also converted the result to a jobject when it is passed back to Java.

We have a few places in JFR where the code seems to take a detour and fetch the oop from the result JavaValue, create a JNI handle, puts it back with set_jobject, and then reads it out with get_jobject. I have a patch where the code simply returns the created JNI handle without installing it into the result JavaValue. I've left that as a separate patch for the JFR team to review. 

There are some usages in C1, but I haven't tried to figure that out.